### PR TITLE
Add custom collective filter to virtual card hosts

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective/issues/4198

Related front-end PR: https://github.com/opencollective/opencollective-frontend/pull/6298

### CODE REVIEW

File: server/graphql/v2/object/Host.js [hunk 1]
```diff
@@ -265,6 +266,16 @@ export const Host = new GraphQLObjectType({
             merchantId = (await fetchAccountWithReference(args.merchantAccount, { throwIfMissing: true })).id;
```
Comment (kewitz) [inline server/graphql/v2/object/Host.js:266]:
```suggestion
            merchantId = (await fetchAccountWithReference(args.merchantAccount, { throwIfMissing: true, loaders: req.loaders })).id;
```

File: server/graphql/v2/object/Host.js [hunk 2]
```diff
@@ -265,6 +266,16 @@ export const Host = new GraphQLObjectType({
...
           }
 
+          let collectiveIds;
+          if (!isEmpty(args.collectiveAccounts)) {
+            const collectives = await Promise.all(
+              args.collectiveAccounts.map(collectiveAccount =>
+                fetchAccountWithReference(collectiveAccount, { throwIfMissing: true }),
```
Comment (kewitz) [inline server/graphql/v2/object/Host.js:273]:
```suggestion
                fetchAccountWithReference(collectiveAccount, { throwIfMissing: true, loaders: req.loaders }),
```

File: server/models/index.js [hunk 3]
```diff
@@ -286,6 +286,7 @@ export function setupModels() {
   });
   m.VirtualCard.hasMany(m.Expense, { foreignKey: 'VirtualCardId', as: 'expenses' });
   m.Collective.hasMany(m.VirtualCard, { foreignKey: 'HostCollectiveId', as: 'virtualCards' });
+  m.Collective.hasMany(m.VirtualCard, { foreignKey: 'CollectiveId', as: 'virtualCardCollectives' });
```
Comment (kewitz) [inline server/models/index.js:289]: Nice!

File: server/graphql/v2/object/Host.js [hunk 4]
```diff
@@ -265,6 +266,16 @@ export const Host = new GraphQLObjectType({
             merchantId = (await fetchAccountWithReference(args.merchantAccount, { throwIfMissing: true })).id;
           }
 
+          let collectiveIds;
```
Comment (kewitz) [inline server/graphql/v2/object/Host.js:269]: :ice_cream: Personally, I'm not a big fan of mutability and I think it is clear to add the conditional when declaring the constant.
```js
const collectiveIds = isEmpty(args.collectiveAccounts) ? undefined : await Promise.all(...).then(collectives => collectives.map(c => c.id))
```

**Review by kewitz — APPROVED** [review]
Awesome! I added a small performance suggestion.

File: server/graphql/v2/object/Host.js (hunk 4 above)
Comment (SudharakaP) [reply server/graphql/v2/object/Host.js]: Done. 👍🏽

**Pull request merged**